chanzuckerberg/idseq-web
# Improve lineagedb rake task

# Description

This improves the usability of this rake task. In following https://github.com/chanzuckerberg/idseq-web/wiki/%5BDev%5D-Index-Update-Instructions, I wanted to understand better the lineage update step. As I played with the code, I made some improvements.

Improvements:
* make version to default to plus one
* make script fail on failed inline shell command `set -e`
* make script print all inline shell commands `set -x`
* add `dryrun` mode
* fix db creds to work in local development (not sure when this broke)
* refactored into two methods
* added some print statements
* added simpler `NCBI_DATE` task param
* added rake task description

# Tests

* run with and without `NCBI_DATE`: `NCBI_DATE=2019-09-17 rails update_lineage_db`
* run with and without dryrun: `NCBI_DATE=2019-09-17 rails update_lineage_db[dryrun]`
* check version increments with dryrun
* force bad database creds and observe early exit
* run to completion once on actual data in local env
